### ISSUE TITLE
Form error states: Improve screen reader readout

### ISSUE THREAD

**Issue by amyleadem** [department-of-veterans-affairs/vets-design-system-documentation#4616]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

> [!note]
> This is possibly related to https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2571, but this issue is more focused on getting the error message to announce when focus moves to the input element. 

## What happened

When form elements like text input or radio enter an error state, the screen reader does not announce the error message, despite the error message span having an alert role. 

## What I expected to happen

When a user submits a form that is missing required values, they should receive immediate feedback that there is an error on the page. However, screen reader users are not notified of this change. 

This is because the alert role only gets activated when the content inside the alert element _updates_, not when it is added to the DOM. From [MDN's docs for alert role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/alert_role): 

> "As with all other [live regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Guides/Live_regions), alerts will only be announced when the content of the element with role="alert" is updated. Make sure that the element with the role is present in the page's markup first - this will "prime" the browser and screen reader to keep watching the element for changes. After this, any changes to the content will be announced. Do not try to dynamically add/generate an element with role="alert" that is already populated with the alert message you want announced - this generally does not lead to an announcement, as it is not a content change."

More information in this [Slack thread](https://wilcore.slack.com/archives/C08N8Q5E9K5/p1754414082205739) (🔒). 

This [test PR](https://github.com/department-of-veterans-affairs/component-library/pull/1720) demonstrates that there are two factors that affect the readout of error messages:
1. The error message span must already be present on the page and receive updates to be announced
2. Focus cannot be simultaneously moved _away_ from the error message. To reliably announce the alert content, the focus must stay on either the button that triggered the error state (Submit/continue button) or move to the error message itself. 

## Reproducing

- CSS Library version:
- Device: Mac
- Browser: Chrome
- Screen readers: VoiceOver, JAWS

Steps to reproduce:

1. Visit [mock form 20-0995](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995/start)
2. Turn on your screen reader
3. Move through the form and trigger an error state in each phase by clicking "Continue" without entering values in the required form fields
4. Notice that the error message appears visually, but does not get announced by the screen reader. 

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

**Comment by amyleadem** [comment]
Adding this issue to this sprint to stay on track with related fixes on the forms site. Details in this [slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1754413828133049) (🔒).
